demos/bubble_shear.py

```diff
@@ -0,0 +1,483 @@
+
+
+# We can now run the simulation over the entire time interval, but now we will adapt the
+# mesh at the beginning of each subinterval using the metric advection algorithm. ::
+
+mesh = UnitSquareMesh(32, 32)
+mesh_numVertices = []
+c = get_initial_condition(mesh)
+for i in range(num_adaptations):
+    t0 = i * interval_length  # subinterval start time
+    t1 = (i + 1) * interval_length  # subinterval end time
+
+    # Adapt the mesh based on the concentration field at t0
+    mesh = adapt_metric_advection(mesh, t0, t1, c)
+    mesh_numVertices.append(mesh.num_vertices())
+
+    c0 = c.copy(deepcopy=True)
+
+    # Solve the advection equation over the subinterval (t0, t1]
+    c = run_simulation(mesh, t0, t1, c)
+
+    plot_mesh(mesh, c0, c, i, "metric_advection")
+
+c0 = get_initial_condition(mesh)
+metric_adv_c0 = Function(FunctionSpace(mesh, "CG", 1)).interpolate(c0)
+metric_adv_error = compute_rel_error(metric_adv_c0, c)
+
+print(
+    f"Metric advection mesh adaptation.\n"
+    f"   Avg. number of vertices: {np.average(mesh_numVertices):.1f}\n"
+    f"   Relative L2 error: {metric_adv_error:.2f}%"
+)
+
+# .. code-block:: console
+#
+#    Metric advection mesh adaptation.
+#       Avg. number of vertices: 2032.7
+#       Relative L2 error: 31.30%
+#
+# .. figure:: bubble_shear-metric_advection_14.jpg
+#    :figwidth: 80%
+#    :align: center
+#
+# The relative L2 error of 31.00% is slightly lower than the one obtained with the
+# classical mesh adaptation algorithm, but note that the average number of vertices is
+# about 15% smaller. Looking into the final adapted mesh and concentration fields in
+# the above figure, we now observe that the mesh is indeed refined in a much wider
+# region - ensuring that the bubble remains well-resolved throughout the subinterval.
+# However, this also means that the available resolution is more widely distributed,
+# leading to a coarser local resolution compared to classical mesh adaptation.
+#
+# Let us now consider the implications and limitations of each approach. Firstly,
+# given a high enough adaptation frequency (i.e. number of adaptations/subintervals),
+# the lagging mesh problem can be mitigated with the classical mesh adaptation
+# algorithm. This may end up being more computationally efficient than the metric
+# advection algorithm, which requires solving the advection equation once more
+# at each timestep, as well as potentially expensive metric computations.
+# However, while increasing the mesh adaptation frequency would alleviate the mesh lag,
+# doing so may introduce large errors due to frequent solution transfers between meshes.
+#
+# This is where the advantage of the metric advection algorithm lies: it predicts where
+# to prescribe fine resolution in the future, and thus avoids the need for frequent
+# solution transfers. We can assure ourselves of that by repeating the above simulations
+# with `num_adaptations = 5`, which yields relative L2 errors of 61.06% and 36.86% for
```

Comment:
Need double backticks to get code format rather than italics.

Reply:
Done in 118de05d0291868951e7469d53a715764e30dc5f